McCorrections/src/EventWeights.cpp

```diff
@@ -47,19 +57,59 @@ EventWeights::EventWeights(Period period, JetOrdering jet_ordering, Discriminato
                         period, jet_ordering, btag_wp);
             else if(jet_ordering == JetOrdering::DeepFlavour)
                 providers[WeightType::BTag] = std::make_shared<BTagWeight>(
-                        FullName("2017/btag/BTagEfficiency_deep_Flavour_pu_id_full.root"), FullName("2017/btag/DeepFlavour_94XSF_V1_B_F.csv"),
+                        FullName("2017/btag/b_eff_HH_DeepFlavour_2017.root"), FullName("2017/btag/DeepFlavour_94XSF_WP_V3_B_F.csv"),
                         period, jet_ordering, btag_wp);
             else if(jet_ordering == JetOrdering::HHJetTag)
                 providers[WeightType::BTag] = std::make_shared<BTagWeight>(
-                        FullName("2017/btag/BTagEfficiency_deep_Flavour_pu_id_full.root"), FullName("2017/btag/DeepFlavour_94XSF_V1_B_F.csv"),
+                        FullName("2017/btag/b_eff_HH_DeepFlavour_2017.root"), FullName("2017/btag/DeepFlavour_94XSF_WP_V3_B_F.csv"),
+                        period, jet_ordering, btag_wp);
+            else
+               throw exception("Jet_Ordering %1% is not supported.") % jet_ordering;
+        }
+        if(mode.empty() || mode.count(WeightType::LeptonTrigIdIso))
+            providers[WeightType::LeptonTrigIdIso] = std::make_shared<LeptonWeights>(
+                        FullLeptonName("Electron/Run2017/Electron_IdIso_IsoLt0.10_eff_RerecoFall17.root"),
+                        FullLeptonName("Electron/Run2017/Electron_Ele32orEle35.root"),
+                        FullLeptonName("Electron/Run2017/Electron_EleTau_Ele24.root"),
+                        FullLeptonName("Muon/Run2017/Muon_IdIso_IsoLt0.15_eff_RerecoFall17.root"),
+                        FullLeptonName("Muon/Run2017/Muon_IsoMu24orIsoMu27.root"),
+                        FullLeptonName("Muon/Run2017/Muon_MuTau_IsoMu20.root"),
+                        FullName("2017/Tau/tauTriggerEfficiencies2017_New.root"),
+                        FullName("2017/Tau/tauTariggerEfficiencies.root"),
+                        period, DiscriminatorWP::Medium,applyTauId);
+                        // POG SFs
+                        // FullName("2017/Electron/EleIdSFPOG.root"),
+                        // FullName("2017/Electron/EleIsoSFPOG.root"),
+                        // FullName("2017/Muon/MuonIdSFPOG.root"),
+                        // FullName("2017/Muon/MuonIsoSFPOG.root"),
+                        // FullName("2017/Muon/MuonTriggerSFPOG.root"),
+                        // FullName("Tau/fitresults_tt_moriond2017.json"),
+    if(mode.empty() || mode.count(WeightType::TopPt))
+        providers[WeightType::TopPt] = std::make_shared<TopPtWeight>(0.0615, 0.0005);
+    }
+
+    else if(period == Period::Run2018) {
+        if(mode.empty() || mode.count(WeightType::PileUp))
+            providers[WeightType::PileUp] = std::make_shared<PileUpWeightEx>(
+                        FullName("2017/DataPileupHistogram_200bin.root"),
```

Comment:
those are the old files, so remove PU, LeptonID and TopPt sections for 2018 for the moment